src/Microsoft.AspNet.Routing/Template/TemplateBinder.cs

```diff
@@ -283,12 +296,22 @@ public string BindValues(RouteValueDictionary acceptedValues)
                     continue;
                 }
 
-                queryBuilder.Add(kvp.Key, converted);
+                if (!wroteFirst)
+                {
+                    context.Writer.Write('?');
+                    wroteFirst = true;
+                }
+                else
+                {
+                    context.Writer.Write('&');
+                }
+
+                _urlEncoder.Encode(context.Writer, kvp.Key);
+                context.Writer.Write('=');
+                _urlEncoder.Encode(context.Writer, converted);
```

Comment:
Replaced usage of the `QueryBuilder`. We can encode directly to the writer, and we can more easily pool this way.